packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1119,22 +1119,25 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
     _updateSourceStream(newStream);
   }
 
-  ImageStreamListener _getListener([ImageLoadingBuilder loadingBuilder]) {
-    loadingBuilder ??= widget.loadingBuilder;
-    _lastException = null;
-    _lastStack = null;
-    return ImageStreamListener(
-      _handleImageFrame,
-      onChunk: loadingBuilder == null ? null : _handleImageChunk,
-      onError: widget.errorBuilder != null
-        ? (dynamic error, StackTrace stackTrace) {
-            setState(() {
-              _lastException = error;
-              _lastStack = stackTrace;
-            });
-          }
-        : null,
-    );
+  ImageStreamListener _imageStreamListener;
+  ImageStreamListener _getListener({bool recreateListener = false}) {
+    if(_imageStreamListener == null || recreateListener) {
+      _lastException = null;
+      _lastStack = null;
+      _imageStreamListener = ImageStreamListener(
+        _handleImageFrame,
+        onChunk: widget.loadingBuilder == null ? null : _handleImageChunk,
+        onError: widget.errorBuilder != null
```

Comment:
I imagine we should turn the anonymous method into a class method and assign it here so that we dont need the recreateListener boolean

Reply:
```suggestion
        onError: widget.errorBuilder != null ? _onError : null
```

Reply:
As I wrote in the OP that is one of the options, but you'd still have to pass all the listeners of the old widget in didUpdateWidget to recreate the listener. (you'd have to know whether or not the errorlistener was null). I don't know why you should recreate the listener every time in the first place.

Reply:
You'd still need the recreate if the widget first doesn't have an error Builder, but the new widget does.

Reply:
ah, that make sense